crates/burn-jit/src/kernel/matmul/simple.rs

```diff
@@ -66,6 +70,8 @@ fn matmul_kernel<F: Float>(
     let unroll_sum = Comptime::map(vectorization_factor, |w: UInt| w != UInt::new(1));
     if Comptime::get(unroll_sum) {
         let mut accum = F::new(0.);
+        // we unroll the loop to sum `vectorization_factor` elements at once, which lets us
```

Comment:
This whole block is simply to handle vectoriation, since when vectorization is enabled, we have `vectorization_factor` different accumulators that need to be summed before stored in the output tensor.

Reply:
yes that's what I was trying to convey by saying that we want so sum `vectorization_factor` at once; feel free to tweak the comment if you feel like it's not accurate